src/webparts/laserficheAdminConfiguration/components/ManageMappingsPage/ManageMappingsPage.tsx

```diff
@@ -450,333 +415,259 @@ export default class ManageMappingsPage extends React.Component<
-            <td>
-              <select
-                name='SharePointContentType'
-                className='custom-select'
-                value={this.state.mappingRows[index].SharePointContentType}
-                id={this.state.mappingRows[index].id}
-                onChange={this.handleChange(index)}
-              >
-                <option>Select</option>
-                {SharePointContents}
-              </select>
-            </td>
-            <td>
-              <select
-                name='LaserficheContentType'
-                className='custom-select'
-                value={this.state.mappingRows[index].LaserficheContentType}
-                id={this.state.mappingRows[index].id}
-                onChange={this.handleChange(index)}
-              >
-                <option>Select</option>
-                {LaserficheContents}
-              </select>
-            </td>
-            <td className='text-center'>
+  const SharePointContents = sharePointContentTypes.map((v) => (
+    <option key={v.name} value={v.name}>{v.name}</option>
+  ));
+  const LaserficheContents = laserficheContentTypes.map((v) => (
+    <option key={v.name} value={v.name}>{v.name}</option>
+  ));
+  const renderTableData = mappingRows.map((item, index) => {
+    if (item.toggle) {
+      return (
+        <tr id='addr0' key={index}>
+          <td>
+            <select
+              name='SharePointContentType'
+              disabled
+              className='custom-select'
+              value={mappingRows[index].SharePointContentType}
+              id={mappingRows[index].id}
+              onChange={(e) => handleChange(e, index)}
+            >
+              <option>Select</option>
+              {SharePointContents}
+            </select>
+          </td>
+          <td>
+            <select
+              name='LaserficheContentType'
+              disabled
+              className='custom-select'
+              value={mappingRows[index].LaserficheContentType}
+              id={mappingRows[index].id}
+              onChange={(e) => handleChange(e,index)}
+            >
+              <option>Select</option>
+              {LaserficheContents}
+            </select>
+          </td>
+          <td className='text-center'>
+            <a
+              href='javascript:;'
+              className='ml-3'
+              onClick={() => EditSpecificMapping(index)}
+            >
+              <span className='material-icons'>edit</span>
+            </a>
+            <a
+              href='javascript:;'
+              className='ml-3'
+              onClick={() => RemoveSpecificMapping(index)}
+            >
+              <span className='material-icons'>delete</span>
+            </a>
+          </td>
+        </tr>
+      );
+    } else {
+      return (
+        <tr id='addr0' key={index}>
+          <td>
+            <select
+              name='SharePointContentType'
+              className='custom-select'
```

Comment:
it seems like this html is very similar to the one above, maybe can refactor

Reply:
made a note of this